app/javascript/mastodon/actions/compose.js

```diff
@@ -174,6 +174,79 @@ export function submitComposeFail(error) {
   };
 };
 
+const MAX_IMAGE_DIMENSION = 1280;
+
+const dataURLtoBlob = dataURL => {
+  const BASE64_MARKER = ';base64,';
+
+  if (dataURL.indexOf(BASE64_MARKER) === -1) {
+    const parts       = dataURL.split(',');
+    const contentType = parts[0].split(':')[1];
+    const raw         = parts[1];
+
+    return new Blob([raw], { type: contentType });
+  }
+
+  const parts       = dataURL.split(BASE64_MARKER);
+  const contentType = parts[0].split(':')[1];
+  const raw         = window.atob(parts[1]);
+  const rawLength   = raw.length;
+
+  const uInt8Array = new Uint8Array(rawLength);
+
+  for (let i = 0; i < rawLength; ++i) {
+    uInt8Array[i] = raw.charCodeAt(i);
+  }
+
+  return new Blob([uInt8Array], { type: contentType });
+};
+
+const resizeImage = (inputFile, callback) => {
+  if (inputFile.type.match(/image.*/)) {
```

Comment:
Animated gifs will pass this test and it will lost animation.

Reply:
But a GIF image larger than 1280px... is there any point in keeping it animated, since it will be a large file size..?

Reply:
but if you do so *silently*. they won't be aware of losing animation before posting it.

I think most cases will be covered even if we exclude GIFs from this process.